source/connection-monitoring-and-pooling/tests/pool-checkout-maxConnecting-is-enforced.yml

```diff
@@ -18,29 +18,30 @@ poolOptions:
   maxPoolSize: 10
   waitQueueTimeoutMS: 5000
 operations:
-  # start creating a Connection. This will take a while
-  # due to the fail point.
+  # start 3 threads
   - name: start
     target: thread1
+  - name: start
```

Comment:
The way the Rust driver runs these tests is that it essentially separates each threads workloads into separate queues and executes them all concurrently, so having the starts happen before the main thread waits will cause all the checkOuts to happen simultaneously. This does seem like an odd way to perform these tests however. I think they're supposed to be run like the main thread dispatching tasks sequentially as they're listed in the format. Is this how Python runs the tests?

Reply:
So it looks like Java, Go, and Node dispatch the threaded tasks in order from the main thread, and Ruby separates them per thread at the beginning and begins execute them one after another once the `start` operation for a given thread is called like Rust does.

I believe the tests were _intended_ to be executed like the former since it makes them much easier to write and read. I'm happy to update the Rust runner to accommodate this, but it is a little worrisome that we'd also make Ruby do the same change. If we think it constitutes a bug in the two implementations, then its probably worth fixing though.

Reply:
After thinking about it some more, I feel like this is a bug in the Ruby and Rust implementations and that drivers should be dispatching threaded tasks from the main thread as opposed to preprocessing them on a per-thread basis, and that we should make this explicit in the spec readme. cc @divjotarora do you agree?

also cc @p-mongo

Reply:
Yeah I agree that the operations should be dispatched in order.

Reply:
Also agreed. I haven't read the spec tests README yet, but it's probably worth updating it to clarify this if we're considering it a bug in those test runners.

Reply:
Yep, @ShaneHarvey could you add that as part of this PR then? I think this would be the first test that actually requires this ordering.

Reply:
Updated, PTAL.

Reply:
looks good to me!